proposals/XXXX-compact.md

```diff
@@ -96,79 +104,167 @@ Compact witness type signatures are a kind of foreign function interface (FFI).
-The mapping from witnesses' Compact type signatures to foreign language type signatures will be predictable and well specified for each supported foreign language.
+The mapping from witnesses' Compact type signatures to foreign language type signatures is predictable and will be well specified for each supported foreign language.
 
 In the specific instantiation of Compact for the Midnight Network, witnesses have TypeScript foreign type signatures and they can be implemented in either TypeScript or JavaScript.
-However, the intention is that other foreign langauges can be equally supported.
+However, the intention is that other foreign languages can be equally well supported.
 
 ## Ledger ADTs
-The public state of a Compact contract is its so-called "ledger".
+The public on-chain state of a Compact contract is its so-called "ledger".
 Compact supports a number of builtin abstract data types (ADTs) for the ledger.
 Examples include cells containing a Compact value, integer counters, sets of Compact values, maps whose keys and values are Compact values, etc.
 
-The specific instantiation of Compact for the Midnight Network has specific ledger ADTs.
+The specific instantiation of Compact for the Midnight Network has specific builtin ledger ADTs.
 However, the design is general enough to allow other ADTs to be provided.
 
 ## The Compact Standard Library
 Compact has a standard library which provides utilities that are generally useful in smart contracts.
 Notable among these are cryptographic primitives such as hashing and elliptic curve arithmetic.
 The ledger ADTs are also implemented by the standard library.
 
+The standard library is extensible by the Compact implementers.
+
+## Developer Tools (the Compact SDK)
+The Compact project includes utilities for Compact contract development.
+These tools are uniformly invoked using the `compact` command-line program.
+This is similar to the way that the `git` command-line program provides uniform access to a large number of subcommands.
+The implementation of the command line driver is in Rust and is part of the Compact open-source project.
+
+There are currently a number of subcommands such as:
+
+- `install` and `upgrade` to manage versions of the compiler and related tooling
+- `compile` to compile Compact contracts
+- `format` to invoke the official Compact source-code formatter
+- `fixup` to make safe automatic repairs to contracts
+
+The design of the SDK tooling is intended for flexibility.
+Subcommands can potentially be implemented in any other language (e.g., Python, shell scripts, etc.) and they could be provided by third parties.
+
 ## The Compiler Implementation
-The Compact project provides an implementation of the Compact programming language.
+The Compact open-source project provides an official implementation of the Compact programming language.
```

Comment:
```suggestion
The Compact open-source project provides a reference implementation of the Compact programming language for Midnight Network.
```

Reply:
I think it's subtly different.

A reference implementation (to me) is intended to explain and illustrate the semantics and to be readable to someone who wants to create other implementations.

An official implementation is the one that is supported by the project (the situation is like the CPython implementation for Python, the official implementation of the Python Software Foundation).  Any others are third party and use at your own risk.